cache/contenthash/checksum.go

```diff
@@ -386,7 +386,7 @@ func (cc *cacheContext) ChecksumWildcard(ctx context.Context, mountable cache.Mo
 
 	wildcards, err := cc.wildcards(ctx, m, p)
 	if err != nil {
-		return "", nil
+		return "", err
```

Comment:
🙄

Reply:
Can we add a test that catches this.

Reply:
I'll try. Not sure how I can force a filesystem failure from the test framework, but I'll see if I can inject something.

Reply:
I ended up having to fake a mount failure, because it turns out the only other possible failure is 'too many symlinks' from `scanPath`, and it can't be triggered unless the root path is a symlink to itself, which causes all kinds of other issues ("." starts getting seen as a path for some reason, and `addParentToMap` blows the stack.